server/src/service/initialize/initialize-service.js

```diff
@@ -80,85 +82,142 @@ async function accountFollowingService(api, id, db) {
+ *
  * @param {MastodonAPI} api - An instance of the MastodonAPI to perform the account related api calls.
- * @param {string} id - Identifier of the account to initialize data for.
+ * @param {string} mainId - Identifier of the account to check for followings.
  * @param {DAO} db - An instance of the DAO to perform database operations.
+ * @param {integer} maxNodes - Max number of nodes to return.
+ * @param {string} nodeRank - Node parameter to rank nodes based on. Options are: "followers", "posts", "random"
+ * @param {integer} locality - Exponential decay of node rank based on distance from mainId. Higher values prioritize closer nodes.
  *
- * @returns {Promise<{relations: object, accountInfoList: object[]}>} Object containing relationship data and a list of account information.
+ * @returns {Promise<array|null>} List of account info objects.
+ *      The 'following' property contains the nodes in this list that the account follows. Null if no data is available.
  */
-async function accountInitializeService(api, id, db) {
-    let relations = {}; // To store relationships between accounts
-    let accountInfoMap = new Map(); // To store account information
-
-    // Initialize the relations for the primary account
-    relations[id] = new Set();
-
-    // Function to process followers and followings
-    async function processAccounts(accounts, isFollowerOfId) {
-        for (const accountId in accounts) {
-            const account = accounts[accountId];
-            accountInfoMap.set(account.id, account); // Add account information to the map
-
-            // Update relations
-            if (!relations[account.id]) {
-                relations[account.id] = new Set();
-            }
-            if (isFollowerOfId) {
-                relations[account.id].add(id); // Add relation as a follower
-            } else {
-                relations[id].add(account.id); // Add relation as following
-            }
+async function accountInitializeService(
+    api,
+    mainId,
+    db,
+    maxNodes = 10,
+    nodeRank = 'followers',
+    locality = 2
+) {
+    let mainNodeInfoRaw;
+    try {
+        mainNodeInfoRaw = await api.getAccountInfo(mainId);
+    } catch (e) {
+        return new Array();
+    }
+    if (!mainNodeInfoRaw || !mainNodeInfoRaw['data']) {
+        return new Array();
+    }
+    const mainNodeInfo = createAccountInfo(mainNodeInfoRaw['data']);
+
+    const seenNodes = new Set([mainId]); // Track seen nodes
+    const accountInfoMap = new Map(); // Track info of selected nodes
+    accountInfoMap.set(mainId, {
+        ...mainNodeInfo,
+        depth: 0,
+        priority: 1,
+    });
+
+    // Sort nodes by priority using custom max heap
+    const heapComparator = (a, b) => b.priority - a.priority;
+    const nodeHeap = new Heap(heapComparator);
+    nodeHeap.init([accountInfoMap.get(mainId)]);
+
+    let rankFun; // Choose parameter to rank nodes by
```

Comment:
I would probably extract this (and even including the calcPriority) into a seperate function - cleaner and eaiser to test (maybe you are already doing it)
Also might be personal preference but I would use switch case for this

Reply:
Yeah I was planning on splitting up the function for unit tests, just didn't get around to it. It's a bit hard bc. there's a lot of variables being using across the function.